testing/docker-master/Dockerfile

```diff
@@ -0,0 +1,13 @@
+FROM ubuntu:xenial
+
+RUN apt update && apt dist-upgrade -y && apt install -y git && apt clean
+
+# enable multiverse as snapcraft cleanbuild does.
+RUN sed -i 's/ universe/ universe multiverse/' /etc/apt/sources.list
+
+# build and install snapcraft from master.
+git clone https://github.com/snapcore/snapcraft
+cd snapcraft
+apt build-dep -y ./
```

Comment:
I'm unsure how apt build-dep works (if they mark the deps as autoinstalled). If so, you can just run `sudo apt autoremove --purge` once you installed your local .deb (so last step).

If not, you can revert to the old way of doing this build-dep install:
https://github.com/ubuntu/ubuntu-make/blob/master/Dockerfile#L53
mk-build-deps, then removing the local .deb for build-deps, and finally apt autoclean as well.

Tell me how it goes :)